base/distributed/cluster.jl

```diff
@@ -136,19 +136,27 @@ worker_timeout() = parse(Float64, get(ENV, "JULIA_WORKER_TIMEOUT", "60.0"))
 
 
 ## worker creation and setup ##
+"""
+    start_worker(out::IO, cookie::AbstractString)
+
+`Base.start_worker` is an internal function which is the default entry point for
+worker processes connecting via TCP/IP. It sets up the process as a Julia cluster
+worker.
+
+`start_worker(cookie)` is equivalent to `start_worker(STDOUT, cookie)`.
+If the cookie is an empty string, `""`, the worker tries to read the cookie from its STDIN.
 
-# The entry point for julia worker processes. does not return. Used for TCP transport.
-# Cluster managers implementing their own transport will provide their own.
-# Argument is descriptor to write listening port # to.
+The function closes STDIN, redirects STDERR to STDOUT, listens on a free port (unless
+the port is explictly specified by the `--bind-to` command line option) and schedules tasks to
+process incoming TCP connections and requests. It does not return.
+"""
 start_worker(cookie::AbstractString) = start_worker(STDOUT, cookie)
 function start_worker(out::IO, cookie::AbstractString)
-    # we only explicitly monitor worker STDOUT on the console, so redirect
-    # stderr to stdout so we can see the output.
-    # at some point we might want some or all worker output to go to log
-    # files instead.
-    # Currently disabled since this caused processes to spin instead of
-    # exit when process 1 shut down. Don't yet know why.
-    #redirect_stderr(STDOUT)
+    if cookie == ""
```

Comment:
Yeah I don't like using `""` to denote that there is no value especially since it might be a perfectly valid cookie. But since it is set from c code it might be too annoying to work with `Nullable`.